### DISCUSSION TITLE
How to call a PHP function with dynamic generated function name?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#978] [Ask a question] [answered]
Hello everyone!
I have a question for the `php` specialists &#128512;

I have a variable that is filled dynamically. This variable contains the function name. How can I call the function?
I thought of `call_user_func()`. Unfortunately the function will not found. I think it is due to the missing `$this->`.

I am looking for something like this but in the right: &#128556;

```
$choosedFilter = "sharpen";
$image = $this->$choosedFilter($image);
```

Does anyone have an idea?

**Comment by PetersOtto** [comment]
Oh, I think I've found the answer myself.

https://www.php.net/manual/en/functions.variable-functions.php

Have a nice weekend!

**Reply by GiovanniSalmeri** [reply, marked as answer]
Also

`call_user_func(array($this, $choosedFilter), $image);`

should do the trick.

**Reply by PetersOtto** [reply]
It do! Thank you &#128512;
